### PR TITLE
Update checking feature

### LINKED ISSUE #694: Check for updates
## Expected Behavior
Check for updates when the application starts and notify the user when there's a new release available to download.

## Possible Solution
Check the [Github API](https://api.github.com/repos/keepassxreboot/keepassxc/releases/latest) for releases and notify the user if there's a new release available.

We can ask the user if he wants to enable this feature on the first run and make it "toggleable" in the settings.

It should be possible to enable/disable this feature with a compiler flag, since package managers have their own updates.

### PR DESCRIPTION
**Pull request by weslly**
[TIP]:  # ( Provide a general summary of your changes in the title above ^^ )

## Type of change
[NOTE]: # ( Please remove all lines which don't apply. )
- ✅ New feature (non-breaking change which adds functionality)

## Description and Context
[NOTE]: # ( Describe your changes in detail, why is this change required? )
[NOTE]: # ( Describe the context of your change. Explain large code modifications. )
[NOTE]: # ( If it fixes an open issue, please add "Fixes #XXX" as necessary )

Add update checking feature using GitHub releases API.

The application ask the user if they want to enable this feature on the first run and make it "toggleable" in the settings.

Fixes #694 

## Screenshots
[TIP]:  # ( Do not include screenshots of your actual database! )
image

image
image
image
image

## Testing strategy
Only added tests for the sem-ver comparing function, since the other tests would require networking.

### Steps to test the API
1. Download the [json from GitHub releases API endpoint](http://api.github.com/repos/keepassxreboot/keepassxc/releases/latest) and rename it to latest.json
2. Create a simple http server with python at the `latest.json` directory: `python -m SimpleHTTPServer 8000`
3. Change the url at UpdateCheck.cpp to http://0.0.0.0:8000/latest.json

## Checklist:
[NOTE]: # ( Please go over all the following points. )
[NOTE]: # ( Again, remove any lines which don't apply. )
[NOTE]: # ( Pull Requests that don't fulfill all [REQUIRED] requisites are likely )
[NOTE]: # ( to be sent back to you for correction or will be rejected.  )
- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ My change requires a change to the documentation, and I have updated it accordingly.
- ✅ I have added tests to cover my changes.

### CODE REVIEW

**Comment by droidmonkey** [comment]
Wow this is fantastic

**Comment by weslly** [comment]
@droidmonkey fixed!

File: src/gui/MainWindow.cpp [hunk 1]
```diff
@@ -661,6 +676,49 @@ void MainWindow::showAboutDialog()
...
+    }
+
+    connect(UpdateCheck::instance(), SIGNAL(updateCheckFinished(bool, QString)), this, SLOT(hasUpdateAvailable(bool, QString)));
+#endif
+}
+
+void MainWindow::hasUpdateAvailable(bool hasUpdate, const QString& version)
+{
+#ifdef WITH_XC_UPDATECHECK
+    disconnect(UpdateCheck::instance(), nullptr, this, nullptr);
```
[the remark this answers is not in the document]
Comment (weslly) [reply src/gui/MainWindow.cpp]: Removing this makes the dialog open twice when clicking "Check for updates", the same signal is already defined on the constructor of UpdateCheckDialog

**Comment by weslly** [comment]
@droidmonkey fixed and rebased 👍

**Comment by weslly** [comment]
@droidmonkey I finished the fixes and added the option to include betas

**Comment by louib** [comment]
@weslly @droidmonkey good job guys!

**Pull request merged**